language/move-compiler/src/naming/fake_natives.rs

```diff
@@ -86,37 +87,24 @@ pub fn resolve_builtin(
     function: &FunctionName,
 ) -> Option<fn(Vec<IR::Type>) -> IR::Bytecode_> {
     let sp!(_, ModuleIdent_ { address, module }) = module;
-    let addr_name = match address {
-        Address::Numerical(Some(sp!(_, n_)), _) | Address::NamedUnassigned(sp!(_, n_)) => n_,
+    // Only resolve if either (a) the address is named "std" or (b) its value is 0x1
+    match address {
+        Address::Numerical(Some(sp!(_, n)), sp!(_, a))
+            if n.as_str() == "std" || a.into_inner() == AccountAddress::ONE => {}
```

Comment:
Ah, interesting case. 

And also interesting that this should have been updated after named addresses were made non-opaque for the packages system. 

Maybe we should also check that `"std"` is bound to `0x1` in the address map? Not really a big deal but just concerned about this being accidental.

Reply:
I thought one idea of aliases is that `std` can be mapped to somewhere else if one desires. In any case, if did not enforce this before, enforcing it now would be a breaking change. So I would just leave as is.